### ISSUE TITLE
Support for Dynamic collection

### ISSUE THREAD

**Issue by nadeemja** [cult-of-coders/grapher#195]
Hi,

My users can `Comment` on `Questions`, `Feedback`, and other `Comments`. 

Every document in the `Comments` collection therefore has a "forCollection" field, so I can know which `Collection` a specific comment-doc has as parent. 

If a user `Comments` on a `Questions`-doc, then that comment-doc will have its `forCollection` field set to `Questions`. Likewise, `forCollection` is set to `Feedback` if a comment is made on a Feedback document.

So, basically, a comment can belong to many different mongo `Collections`.
How to use Grapher in this situation?

**Comment by theodorDiaconu** [comment]
create multiple links:

commentId, questionId, feedbackId inside the comment, and in their inversed side as well for easy fetching.

**Comment by theodorDiaconu** [comment]
Support for Dynamic Collection will not happen, because we need to know the collection prior to building the Graph, otherwise, we had into the same performance problems GraphQL/Apollo currently has.

**Comment by nadeemja** [comment]
I think you are correct :)

**Comment by Herteby** [comment]
I suppose you could technically also overload the same field, like this:
```
Comments.addLinks({
  question:{
    collection:Questions,
    field:'link_id'
  },
  feedback:{
    collection:Feedback,
    field:'link_id'
  },
  comment:{
    collection:Comment,
    field:'link_id'
  }
})
```
Because you can create quadrillions of random Mongo IDs before you get a duplicate.
It might give worse performance though, because the `_id:{$in:_ids}` that I'm guessing Grapher uses would get a much bigger array of _ids?

**Comment by theodorDiaconu** [comment]
@Herteby I think that using your approach can also work, and you're right the .find() will be heavier because it will search through more 'link_id' s when trying to fetch from inversed side.
